rs/https_outcalls/consensus/src/pool_manager.rs

```diff
@@ -364,6 +364,12 @@ impl CanisterHttpPoolManagerImpl {
             .with_label_values(&["generate_change_set"])
             .start_timer();
         let mut change_set = Vec::new();
+
+        // Whenever we have artifacts to purge, we insert the purge change actions before everything
+        // else, to avoid having in the validated pool artifacts belonging to different epochs and
+        // hence preserving the expected maximal number of artifacts in the pool.
+        change_set.extend(self.purge_shares_of_processed_requests(canister_http_pool));
```

Comment:
@eichhorl another possibility here could be to have an early return as soon as we detect some purging backlog. This would not make any assumptions on the ordering of change action execution, but potentially slightly less efficient.

I couldn't justify this, so I propose the most simple change (but have no strong feeling in either case).